arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/Bracket.kt

```diff
@@ -13,6 +13,11 @@ public sealed class ExitCase {
 
   public data class Cancelled(val exception: CancellationException) : ExitCase()
   public data class Failure(val failure: Throwable) : ExitCase()
+
+  public companion object {
+    public fun ExitCase(error: Throwable): ExitCase =
```

Comment:
I was wondering if we could name this `fromThrowable`, so it is aligned with the naming conventions in other parts of Arrow?

Reply:
I was wondering about this too. Do we currently have `fromThrowable` anywhere? Or is the in analog to `Option.fromNullable`?

I'm not sure if that's a great name either, but I cannot think of a similar construct in the Kotlin Std.

Reply:
> I'm not sure if that's a great name either, but I cannot think of a similar construct in the Kotlin Std.
I meant to edit the Kotlin Std lib part it away 😅

Yes it would be analog to `Option.fromNullable`, but maybe the top-level `exitCase` would suffice wdyt @nomisRev ?

Reply:
I.e.
 - [`Result`](https://github.com/JetBrains/kotlin/blob/ea836fd46a1fef07d77c96f9d7e8d7807f793453/libraries/stdlib/src/kotlin/util/Result.kt#L92) is `success` and `failure` on the `Companion`.
 - [Duration.kt](https://github.com/JetBrains/kotlin/blob/ea836fd46a1fef07d77c96f9d7e8d7807f793453/libraries/stdlib/src/kotlin/time/Duration.kt#L113) has `milliseconds`, `seconds` as extensions on `Int` inside the `Companion` 🤯 
 - [Random](https://github.com/JetBrains/kotlin/blob/ea836fd46a1fef07d77c96f9d7e8d7807f793453/libraries/stdlib/src/kotlin/random/Random.kt#L318) top-level constructors
 
 Don't think we have access to `@InlineOnly` though.
 
 Some alternatives:
  - (Companion, Result) `fun throwable(error: Throwable): ExitCase`
  - (Companion, Duration) `fun Throwable.toExitCase(): ExitCase`
  - (top-level, Random) `fun ExitCase(error: Throwable): ExitCase`

PS: @i-walker I think we use `fromNullable` because we need to explicitly distinct between the constructor that allows `Option<A?>` and `Option<A>`, otherwise you can never have `Some(null)` which you want in generic cases.

Reply:
>  I think we use fromNullable because we need to explicitly distinct between ...

Gotcha !

And the alternatives look good to me :D

Reply:
Not quite sure where we are with this. I'm happy to make changes

Reply:
Sorry, @custommonkey.

We were wondering if there is a convention for such APIs in Kotlin, since we've not established a clear convention within Arrow. Looking at the Kotlin Std APIs it doesn't _seem_ to be consistent within the Std either..

TBH I like your proposal because it works well in this case.

`Option` can follow the same guidelines where we can place a `fun Option(...)` constructor on the companion to replace `fromNullable`, and we can keep `null.toOption()` as well.
This would then also belong on the `Companion` similar to how it's encoded for `Duration.kt`.

WDYT @i-walker?
If Imran agrees, there is no action required from you @custommonkey and we'll go ahead with merging the PR as is.

Reply:
Ok, let's keep it how it is. Maybe down the line we could draft a standard for 2.0, but this has low priority in comparison to other efforts in Arrow